src/lib/address_resolve/AddressResolve.h

```diff
@@ -247,6 +247,11 @@ class Resolver
     /// a clear decision if the callback should or should not be invoked.
     virtual CHIP_ERROR CancelLookup(Impl::NodeLookupHandle & handle, FailureCallback cancel_method) = 0;
 
+#if CHIP_DEVICE_ENABLE_CASE_DNS_CACHE
+    // cache node address
```

Comment:
This comment restates what the function name is. Could we update it to explain:

- what is a a cache expected to be? How does it interact? Do we expect cached nodes to be immediately visible in lookups? How is the cache invalidated? What is the expected functionality that a resolver needs to provide? What kind of errors are expected - caches are generally transient so an error seems odd here.